File: ckgame-java/src/test/java/com/ckgame/game/ScenarioTest.java
package com.ckgame.game;

import com.ckgame.core.Constants;
import com.ckgame.world.County;
import com.ckgame.world.World;
import org.junit.jupiter.api.Test;

import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

/** 1066 场景加载不变量，与 Python/Go/TS 版同口径。 */
class ScenarioTest {

    @Test
    void scenarioCounts() {
        World w = ScenarioLoader.loadScenario();
        assertEquals(25, w.characters.size());
        assertEquals(14, w.map.counties().size());
        assertEquals(19, w.titles.size());
        assertEquals(7, w.dynasties.size());
    }

    @Test
    void startDateIs10660101() {
        World w = ScenarioLoader.loadScenario();
        assertEquals(1066, w.date.year());
        assertEquals(1, w.date.month());
        assertEquals(1, w.date.day());
    }

    @Test
    void everyCountyHasHolderAndTitle() {
        World w = ScenarioLoader.loadScenario();
        for (Map.Entry<Integer, County> e : w.map.counties().entrySet()) {
            County c = e.getValue();
            assertNotEquals(Constants.NONE_ID, c.holder, c.name + " 无领主");
            assertNotEquals(Constants.NONE_ID, c.ownerTitle, c.name + " 无所属头衔");
            assertTrue(w.titles.containsKey(c.ownerTitle), c.name + " 的头衔不存在");
        }
    }
}

File: ckgame-ts/src/test/Scenario.test.ts
import { test } from 'node:test';
import assert from 'node:assert/strict';
import { ScenarioLoader } from '../game/ScenarioLoader.js';

const world = ScenarioLoader.loadScenario();

test('1066 场景数据规模', () => {
  assert.equal(world.characters.size, 25);
  assert.equal(world.map.counties.size, 14);
  assert.equal(world.titles.size, 19);
  assert.equal(world.dynasties.size, 7);
});

test('起始日期为 1066-01-01', () => {
  assert.equal(world.date.year(), 1066);
  assert.equal(world.date.month(), 1);
  assert.equal(world.date.day(), 1);
});

test('每个省份都有领主与所属头衔', () => {
  for (const county of world.map.counties.values()) {
    assert.notEqual(county.holder, 0, `${county.name} 无领主`);
    assert.ok(county.ownerTitle, `${county.name} 无所属头衔`);
  }
});

test('存在贸易路线', () => {
  const routes = (world as any).tradeRoutes;
  assert.ok(Array.isArray(routes) && routes.length >= 1);
});
